ccc/clamav.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed
+# under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import tarfile
+import typing
+
+import util
+
+from model.clamav import ClamAVConfig
+from clamav.client.api import ClamAVApi
+from clamav.client.model import ClamAVScanResult
+from clamav.client.routes import ClamAVRoutes
+from container.registry import retrieve_container_image
+
+logger = logging.getLogger(__name__)
+
+
+class ClamAVClient(object):
+    def __init__(
+        self,
+        clamav_config: ClamAVConfig,
+    ):
+        url = f"http://{clamav_config.cluster_url()}"
+        routes = ClamAVRoutes(base_url=url)
+        self._api = ClamAVApi(routes=routes)
+
+    @staticmethod
+    def from_config_name(
+        clamav_config_name: str,
+    ):
+        cfg_factory = util.ctx().cfg_factory()
+        clamav_config = cfg_factory.clamav(clamav_config_name)
+        return ClamAVClient(clamav_config=clamav_config)
+
+    def scan_container_image(self, image_reference: str) -> typing.Tuple[ClamAVScanResult, str]:
```

Comment:
i think this method should either be mv'ed to clamav.client.api (which might be better-named client). ccc should just contain factory functions/methods

although admittedly, we are not 100% consistent in this regard. e.g. elasticsearch and secrets-server both "live" here. but otoh they are fairly small and do not have additional modules.